lint/testdata/lostSQLCloseForRows/negative_tests.go

```diff
@@ -0,0 +1,144 @@
+package checker_tests
+
+import "database/sql"
+
+func normalLocalUseWithDefer() {
+    db, err := sql.Open("postgres", "")
+    if err != nil {
+        return
+    }
+
+    rows, err := db.Query("SELECT * FROM testtable")
+    if err != nil {
+        return
+    }
+    defer rows.Close()
+
+    for rows.Next() {
+        var testdata string
+        rows.Scan(&testdata)
+    }
+
+    return
+}
+
+func normalLocalUseWithoutDefer() {
+    db, err := sql.Open("postgres", "")
+    if err != nil {
+        return
+    }
+
+    rows, err := db.Query("SELECT * FROM testtable")
+    if err != nil {
+        return
+    }
+
+    for rows.Next() {
+        var testdata string
+        rows.Scan(&testdata)
+    }
+    rows.Close()
+
+    return
+}
+
+func normalUseWhereRowsFromOtherMethodWithDefer() {
+    db, err := sql.Open("postgres", "")
+    if err != nil {
+        return
+    }
+
+    rows, err := testMethodReturningRows(db)
+    if err != nil {
+        return
+    }
+    defer rows.Close()
```

Comment:
Can we add another example, when defer is inside an anonymous function?
```
defer func() {
    rows.Close()
}()
```

Reply:
Ok. It is possible...

Reply:
I create negative_tests for this case, but temporary comment. For feature release.